.github/workflows/tests.yml

```diff
@@ -34,7 +34,7 @@ jobs:
     - name: Run unit and C tests
       run: make check -j 2
 
-  ubuntu-latest:
+  ubuntu-latest-gcc:
     runs-on: ubuntu-latest
```

Comment:
Why did you just change the ubuntu-latest and not introduced a new CI run?

Reply:
Because I felt it wasteful to run three compilations in parallel that basically does the same thing.
It should be easy to determine if it is due to g++ or OS version that cause the issue, in case it fails.